### PR TITLE
block_client_connection

### PR DESCRIPTION
**Pull request by iZIVer**
* added blocking connection to the server and set a timeout of 10 sec

### CODE REVIEW

File: grpc/go-cubic/client.go [hunk 1]
```diff
@@ -42,6 +43,8 @@ type Client struct {
 	streamingBufSize uint32
 }
 
+const connectionTimeout = 10*time.Second
```
Comment (happyalu) [inline grpc/go-cubic/client.go]: could you please add an option to configure this value in NewClient?

Comment (jhollowayj) [reply grpc/go-cubic/client.go]: Something like this:

// WithDialTimeout returns an Option which sets a timeout value to be used during the gRPC Dial step.
func WithDialTimeout(timeout time.Duration) Option {
	return func(c *Client) error {
		c.dialTimeout = timeout
		return nil
	}
}

**Comment by iZIVer** [comment]
> Please add tests to check that this works (client_test.go):
> 
> * calling NewClient with an invalid URL should fail now, even if it might have worked earlier.
> * calling NewClient without Insecure() on a valid URL should fail now,
> * calling NewClient on a valid URL suceeds.

Done

**Comment by happyalu** [comment]
@iZIVer sorry I couldn't get to this earlier. This looks good, thank you for the update. The new tests look good, but there seem to be some issues with the earlier tests, perhaps unrelated to WithBlock. I will try to look into that and add commits to this branch.

File: grpc/go-cubic/client_test.go [hunk 2]
```diff
@@ -277,6 +278,28 @@ func TestStreamingBufSize(t *testing.T) {
 
 }
 
+func TestClient_InvalidURL(t *testing.T) {
+	if _, err := cubic.NewClient(fmt.Sprintf("wrong_localhost:2727"), cubic.WithInsecure(),
+		cubic.WithConnectTimeout(200*time.Millisecond)); err == nil {
+		t.Errorf("connecting to invalid server: want error, got nil")
+	}
```
[the remark this answers is not in the document]
Comment (happyalu) [reply grpc/go-cubic/client_test.go]: I agree we can make the tests more readable throughout all the files. I'll defer it to a separate PR.

**Pull request merged**